### PR TITLE
D bindings for ArrayFire using C APIs.

### PR DESCRIPTION
**Pull request by prasunanand**
* Created D bindings for ArrayFire.
* D doesn't have complete support binding through C++.
* Swig was not an alternative as it has no proper support especially for arrays.
* Works perfectly with `LDC` for linear algebra, which I am concerned with.

Please review :=) .

I will add the examples in next PR.

### CODE REVIEW

File: dub.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+	"name": "arrayfire",
+	"authors": [
+		"Prasun Anand"
+	],
+  "libs": ["afcuda", "cusolver", "cudart", "cufft", "cublas"],
```
Comment (pavanky) [inline dub.json]: What is the purpose of this ?

**Review by pavanky — COMMENTED** [review]
Can you remove all the comments if possible ?

File: dub.json (hunk 1 above)
Comment (prasunanand) [reply dub.json]: Ok, this line is wrong. Removing it.

Comment (prasunanand) [reply dub.json]: @pavanky , Don't we need the [documentation](https://dlang.org/spec/ddoc.html) for `arraytfire-d`? I had left the comments there for documentation.

**Comment by prasunanand** [comment]
@pavanky : I left the comments for docs. What do you suggest?

**Comment by pavanky** [comment]
@prasunanand This is not really a "D" binding. This is a good first step, it exports the C API so it can be used from D. 

We still need to implement a D `array` class similar to what we have in C++ and wrap all the functions to be more D like.

**Comment by prasunanand** [comment]
Cool. I will remove these comments.

I would look into implementing a D array class similar to C++  making it more D like . 
Maybe we can have it in the next PR :) .

**Comment by pavanky** [comment]
Sure, remove the comments when you are adding the class. I am merging this in for now.

**Comment by prasunanand** [comment]
Thank You :) 
I had started working on removing the comments btw for this merge itself.

**Pull request merged**